SimG4CMS/PPS/src/PPSPixelSD.cc

```diff
@@ -0,0 +1,437 @@
+// -*- C++ -*-
+//
+// Package:     PPS
+// Class  :     PPSPixelSD
+//
+// Implementation:
+//     <Notes on implementation>
+//
+// Original Author: F.Ferro
+//         Created:  May 4, 2015
+//
+
+// system include files
+
+// user include files
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "SimG4Core/Notification/interface/TrackInformation.h"
+#include "SimG4Core/Notification/interface/G4TrackToParticleID.h"
+#include "SimG4Core/Physics/interface/G4ProcessTypeEnumerator.h"
+
+#include "SimDataFormats/TrackingHit/interface/UpdatablePSimHit.h"
+#include "SimDataFormats/SimHitMaker/interface/TrackingSlaveSD.h"
+
+#include "SimG4CMS/PPS/interface/PPSPixelSD.h"
+#include "SimG4CMS/PPS/interface/PPSPixelNumberingScheme.h"
+
+#include "G4SDManager.hh"
+#include "G4Step.hh"
+#include "G4Track.hh"
+#include "G4VProcess.hh"
+
+#include "G4PhysicalConstants.hh"
+#include "G4SystemOfUnits.hh"
+
+PPSPixelSD::PPSPixelSD(const std::string& name,
+                       const edm::EventSetup& es,
+                       const SensitiveDetectorCatalog& clg,
+                       edm::ParameterSet const& p,
+                       SimTrackManager const* manager)
+    : SensitiveTkDetector(name, es, clg, p),
+      numberingScheme(nullptr),
+      hcID(-1),
+      theHC(nullptr),
+      theManager(manager),
+      currentHit(nullptr),
+      theTrack(nullptr),
+      currentPV(nullptr),
+      unitID(0),
+      previousUnitID(0),
+      preStepPoint(nullptr),
+      postStepPoint(nullptr),
+      eventno(0) {
+  //Add PPS Sentitive Detector Names
+  collectionName.insert(name);
+
+  //Parameters
+  edm::ParameterSet m_p = p.getParameter<edm::ParameterSet>("PPSPixelSD");
+  int verbn = m_p.getUntrackedParameter<int>("Verbosity");
+  verbn = 10000;
+  SetVerboseLevel(verbn);
+  slave = new TrackingSlaveSD(name);
```

Comment:
should be a unique_ptr constructed with make_unique

Reply:
Done, as well as the redefinition of the verbosity level (line 61) removed